src/ui/map.js

```diff
@@ -1043,20 +1047,43 @@ class Map extends Camera {
      * @see [Example: Display a web map using an alternate projection](https://docs.mapbox.com/mapbox-gl-js/example/map-projection/)
      * @see [Example: Use different map projections for web maps](https://docs.mapbox.com/mapbox-gl-js/example/projections/)
      */
-    setProjection(projection?: ?ProjectionSpecification | string) {
+    setProjection(projection: ProjectionSpecification | string = {name: "mercator"}) {
         this._lazyInitEmptyStyle();
         if (typeof projection === 'string') {
             projection = (({name: projection}: any): ProjectionSpecification);
         }
-        this._runtimeProjection = projection;
-        this._transitionFromGlobe = false;
-        this.style.updateProjection();
+        this.updateProjection(projection);
         return this;
     }
 
-    _setImplicitProjection(projection: ProjectionSpecification) {
-        this._runtimeProjection = projection;
-        this.style.updateProjection();
+    updateProjection(projection?: ProjectionSpecification) {
```

Comment:
As far as I understand, `updateProjection` isn't supposed to be exposed in a public API — we have `setProjection` for this, so `_updateProjection` would be a more fitting name. But looking further, at this point there doesn't seem to be much value in keeping these separate — `setProjection` is now just a wrapper around `updateProjection` that additionally handles the first call without a style (`this._lazyInitEmptyStyle()` will be a noop on subsequent calls), and string alias. So we might as well just inline `updateProjection` into `setProjection` and use the latter internally across the codebase?

Reply:
Good idea @Vlad!

Reply:
I've combined `updateProjection` and `setProjection` but it's breaking `null` expression fallback behavior. (This is the reason for the current failing test and is described [here](https://github.com/mapbox/mapbox-gl-js/pull/11204). Calls from the API need to save the  result of `transform.setProjection` as a high priority "explicit" expression while other projection updates (globe transition or style spec changes) should not. IMO the simplest way to maintain this is to keep the two as separate functions.